### PR TITLE
Deploy script

### PR DESCRIPTION
**Pull request by snreynolds**
## Related Issue

Which issue does this pull request resolve?

### CODE REVIEW

File: broadcast/DeployPoolManager.s.sol/11155111/run-latest.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  "transactions": [
+    {
+      "hash": "0xccf3b782e02a69846f39ab3cc0b899e1930c33997c7a1cc1099991894eeb3c9e",
+      "transactionType": "CREATE",
+      "contractName": "PoolManager",
+      "contractAddress": "0xc021a7deb4a939fd7e661a0669fab5ac7ba2d5d6",
```
Comment (hensha256) [inline broadcast/DeployPoolManager.s.sol/11155111/run-latest.json:7]: i definitely dont think we should be merging 4 broadcast files for each chain? some of these are deploying to another address that isnt verified etc?

File: broadcast/DeployPoolManager.s.sol/11155111/run-1717160617.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
+{
+  "transactions": [
+    {
+      "hash": "0x92abf70e87259e80673a3e7b9f665ac26bdc5d74e7c6e7eb3acc6670333ae04b",
+      "transactionType": "CREATE",
+      "contractName": "PoolManager",
+      "contractAddress": "0x69543078d9aa5c4fa3fadbbf9e1d9ad802254a20",
```
Comment (hensha256) [inline broadcast/DeployPoolManager.s.sol/11155111/run-1717160617.json:7]: e.g. this address?

Comment (snreynolds) [reply broadcast/DeployPoolManager.s.sol/11155111/run-1717160617.json:7]: for sure i just added these for now - i can delete before we merge

File: script/01_PoolManager.s.sol [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import "forge-std/console2.sol";
+
+contract DeployPoolManager is Script {
+    function setUp() public {}
+
+    function run(uint256 controllerGasLimit) public returns (IPoolManager manager) {
+        vm.startBroadcast();
+
+        manager = new PoolManager(controllerGasLimit);
```
Comment (shuhuiluo) [inline script/01_PoolManager.s.sol:17]: It'd be cool to have a universal vanity address across chains using `CREATE2` or `CREATE3` deployment.

Comment (hensha256) [reply script/01_PoolManager.s.sol:17]: Thats the plan :) this is just a temporary script for our testnet deployments

File: script/02_PoolModifyLiquidityTest.s.sol [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import {PoolModifyLiquidityTest} from "../src/test/PoolModifyLiquidityTest.sol";
+
+contract PoolModifyLiquidityTestScript is Script {
+
+    address poolManager = 0xc021A7Deb4a939fd7E661a0669faB5ac7Ba2D5d6;
+    function setUp() public {}
+
+    function run() public {
+        vm.broadcast();
+        new PoolModifyLiquidityTest(IPoolManager(poolManager));
```
Comment (hensha256) [inline script/02_PoolModifyLiquidityTest.s.sol:15]: console log address

File: script/02_PoolModifyLiquidityTest.s.sol [hunk 5]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import {Script, console2} from "forge-std/Script.sol";
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {PoolModifyLiquidityTest} from "../src/test/PoolModifyLiquidityTest.sol";
+
+contract PoolModifyLiquidityTestScript is Script {
+
+    address poolManager = 0xc021A7Deb4a939fd7E661a0669faB5ac7Ba2D5d6;
```
Comment (hensha256) [inline script/02_PoolModifyLiquidityTest.s.sol:10]: would be ideal to pass this in? maybe its best just to have 1 script called TestnetDeployScript or something where all of them get deployed and pass the pm between them

File: script/03_PoolSwapTest.s.sol [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import {IPoolManager} from "../src/interfaces/IPoolManager.sol";
+import {PoolSwapTest} from "../src/test/PoolSwapTest.sol";
+
+contract PoolSwapTestScript is Script {
+    address poolManager = 0xc021A7Deb4a939fd7E661a0669faB5ac7Ba2D5d6;
+    function setUp() public {}
+
+    function run() public {
+        vm.broadcast();
+        new PoolSwapTest(IPoolManager(poolManager));
```
Comment (hensha256) [inline script/03_PoolSwapTest.s.sol:14]: same comments here as MLTest

**Review by hensha256 — CHANGES_REQUESTED** [review]

**Comment by snreynolds** [comment]
closing in favor of https://github.com/Uniswap/v4-periphery/pull/293

**Pull request closed without merging**